### PR TITLE
Fix bug where the parameter value is too large to be passed to the python operators

### PR DESCRIPTION
**Pull request by kenxu95**
## Describe your changes and why you are making these changes

During dag construction, we write the parameter value to a path and pass that path in through the spec. The parameter operator now expects that the output path has already been populated, and relegates itself to simply writing metadata and enforcing types.

This solution does not rely on any DB changes, but it does mean that for every run of a flow, we will write the parameter values to storage before calling orchestration.

@saurav-c to validate that this will work with airflow.

## Related issue number (if any)
ENG-1565

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Comment by kenxu95** [comment]
@cw75 @HarisChoudhary I'd like to write an integration test to validate that this fixes the "parameter-value-too-large" problem. Do either of you have the image data that was causing this issue? Any idea how big is too big?

**Comment by HarisChoudhary** [comment]
> @cw75 @HarisChoudhary I'd like to write an integration test to validate that this fixes the "parameter-value-too-large" problem. Do either of you have the image data that was causing this issue? Any idea how big is too big?

The two image i used to discover the issue. One is a smaller resized version that should work, the larger one will error out.

wss-aqueduc
![aqimg
t-roman-france_0](https://user-images.githubusercontent.com/6863633/190524341-4e91daf5-4fe2-45c1-8a28-323b64ac59fb.jpeg)

**Comment by kenxu95** [comment]
cool - I kinda think that instead of adding this to the repo, we can have this in the S3 bucket that we can pull from in our SDK integration tests. https://linear.app/aqueducthq/issue/ENG-1635/build-s3-support-into-our-sdk-integration-tests

Can add it as a note to that task. I'd love to get to this soon actually - there's a bunch of improvements we can make to our suite.

**Comment by HarisChoudhary** [comment]
> cool - I kinda think that instead of adding this to the repo, we can have this in the S3 bucket that we can pull from in our SDK integration tests. https://linear.app/aqueducthq/issue/ENG-1635/build-s3-support-into-our-sdk-integration-tests
> 
> Can add it as a note to that task. I'd love to get to this soon actually - there's a bunch of improvements we can make to our suite.

I dont think we need to test with the image specifically, when passed into the executor its a base64 encoded byte buffer of the image which comes out to a fairly large string. If we can just add a string of that fairly large size then we can replicated this in test without the image needed.

File: src/golang/lib/engine/aq_engine.go [hunk 1]
```diff
@@ -659,7 +656,7 @@ func (eng *aqEngine) TriggerWorkflow(
 	workflowId uuid.UUID,
 	name string,
 	timeConfig *AqueductTimeConfig,
-	parameters map[string]param.Param,
+	parameters map[string]string,
```
Comment (HarisChoudhary) [inline src/golang/lib/engine/aq_engine.go:659]: I don't think this will compile

**Review by HarisChoudhary — CHANGES_REQUESTED** [review]
Seems like maybe some changes were on local that didnt get reflected on the pr but the current state wont work as is i think

File: src/golang/lib/engine/aq_engine.go (hunk 1 above)
Comment (kenxu95) [reply src/golang/lib/engine/aq_engine.go:659]: oh yes I think I checked out the wrong branch's file here

**Review by saurav-c — COMMENTED** [review]
This looks compatible with Airflow

**Review by HarisChoudhary — APPROVED** [review]

**Pull request merged**